sdk/kiosk/src/bcs.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { bcs } from '@mysten/sui.js';
+
+/**
+ * The Kiosk object fields (for BCS queries).
+ */
+export type Kiosk = {
+  id: string;
+  profits: string;
+  owner: string;
+  itemCount: number;
+  allowExtensions: boolean;
+};
+
+// Register the `Kiosk` struct for faster queries.
+bcs.registerStructType('Kiosk', {
```

Comment:
As an aside, I'm fine that this uses BCS directly, but we really do need to start investing in making this package better so that it's saner to use in cases like this (e.g. not using a global registration namespace, auto-generated types, tooling, etc).